drivers/pwm/pwm-ts.c

```diff
@@ -90,49 +83,36 @@ static int ts_pwm_calc(struct ts_pwm *ts,
 
 		dev_dbg(ts->chip.dev, "shift=%u cnt=%u duty_cnt=%u\n",
 			shift, cnt, duty_cnt);
-
-		ts->shift = shift;
-		ts->period = cnt;
-		ts->duty = cnt - duty_cnt;
+		duty_reg = cnt - duty_cnt;
 	}
 
+	writew(cnt, ts->base + REG_PERIOD);
+	writew(duty_reg, ts->base + REG_DUTY);
+	writew(shift, ts->base + REG_SHIFT);
+
 	return 0;
 }
 
 static int ts_pwm_apply(struct pwm_chip *chip, struct pwm_device *pwm,
 			const struct pwm_state *state)
 {
 	struct ts_pwm *ts = to_ts_pwm(chip);
-	u16 value = 0;
 	int err;
-
-	BUG_ON(!state);
-
-	if (!state->enabled) {
-		writel(0x0, ts->base + REG_CONFIG);
-		ts->state.enabled = false;
-		return 0;
-	}
+	u16 ctrl = 0;
 
 	if (state->polarity == PWM_POLARITY_NORMAL)
-		value &= ~(INVERSED);
+		ctrl &= ~INVERSED;
 	else
-		value |= INVERSED;
+		ctrl |= INVERSED;
```

Comment:
This can be just
```
if (state->polarity != PWM_POLARITY_NORMAL)
		ctrl |= INVERSED;
```

No need for the if/else

Reply:
Good point, will fix.